tests/matchers/test_iou.py

```diff
@@ -15,146 +15,146 @@
+
+        # Low threshold, one to one, only one matches
+        gtcells, rescells = _match_nodes(gt, pred, threshold=0.3, one_to_one=True)
+        computed_matches = list(zip(gtcells, rescells))
+        assert ((1, 3) in computed_matches) != ((2, 3) in computed_matches)
+
+    @pytest.mark.parametrize(
+        "data_fxn", [ex_segs.multicell_2d, ex_segs.multicell_3d], ids=["2D", "3D"]
+    )
+    def test_multiple_objects(self, data_fxn):
+        gt, pred = data_fxn()
+        expected_matches = [(1, 3)]
+        gtcells, rescells = _match_nodes(gt, pred)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+    @pytest.mark.parametrize(
+        "data_fxn", [ex_segs.no_overlap_2d, ex_segs.no_overlap_3d], ids=["2D", "3D"]
+    )
+    def test_no_overlap(self, data_fxn):
+        gt, pred = data_fxn()
+        expected_matches = []
+        gtcells, rescells = _match_nodes(gt, pred)
+        assert Counter(expected_matches) == Counter(list(zip(gtcells, rescells)))
+
+    def test_input_error(self):
+        im = np.zeros((10, 10))
+        with pytest.raises(ValueError):
+            # Test that threshold 0 is not valid when not one-to-one
+            gtcells, rescells = _match_nodes(im, im, threshold=0.0)
 
-    with pytest.raises(ValueError):
-        # Test that threshold 0 is not valid when not one-to-one
-        gtcells, rescells = _match_nodes(im1, im2, threshold=0.0)
-
-
-def test__match_nodes_threshold():
-    im1, im2 = get_two_to_one(10, 10, 30, 30)
-    # Test high threshold
-    gtcells, rescells = _match_nodes(im1, im2, threshold=1)
-    # Create match tuples
-    matches = list(zip(gtcells, rescells))
-    # Check that nothing is matched
-    assert len(matches) == 1
-
-    # Test for high threshold and one to one
-    gtcells, rescells = _match_nodes(im1, im2, threshold=0.7, one_to_one=True)
-    # Create match tuples
-    matches = list(zip(gtcells, rescells))
-    # Check that nothing is matched
-    assert len(matches) == 1
-
-
-def test__match_nodes_non_sequential():
-    # test when the segmentation ids are high numbers (the lower numbers should never appear)
-
-    im1, im2 = get_no_overlap(30, 30)
-
-    # Test that phantom segmentations are not matched
-    gtcells, rescells = _match_nodes(im1, im2, threshold=0.1)
-    # Create match tuples
-    matches = list(zip(gtcells, rescells))
-    # Check that nothing is matched
-    assert len(matches) == 0
-
-    # Test that with one-to-one, phantom segmentations are not matched,
-    # even with threshold 0
-    gtcells, rescells = _match_nodes(im1, im2, threshold=0.0, one_to_one=True)
-    # Create match tuples
-    matches = list(zip(gtcells, rescells))
-    # Check that nothing is matched
-    assert len(matches) == 0
+    @pytest.mark.parametrize(
+        "data_fxn", [ex_segs.no_overlap_2d, ex_segs.no_overlap_3d], ids=["2D", "3D"]
+    )
+    def test_non_sequential(self, data_fxn):
```

Comment:
Just for my understanding, I'm not sure what this is testing? Why does it matter if the segmentation IDs are high non-sequential numbers?

Reply:
There were more comments on the old version - I think we had issues at some point where dummy nodes from the match matrix were appearing in the matched output even though they weren't in the graph, since we added rows in the matrix up to the highest node id and used the index to represent the id. Not sure if we still do that or not, since I know we made the implementation more efficient.

Reply:
Caroline is correct about the issue we were trying to catch. I can add back in some comments so it's clearer. I was inclined to leave it in as a test case since it was an issue in the past even if we may have eliminated the opportunity for the problem.

Reply:
Ok great, thanks for the extra context. Agreed a short docstring would be good, can even link to this comment chain for posterity